### PR TITLE
storybook added

### PR DESCRIPTION
**Pull request by ufoBeliever**
storybook added

closing issue https://github.com/igorizviekov/mt-booking_app/issues/18

### CODE REVIEW

File: src/components/InputComponent/InputComponent.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export const Default = Template.bind({});
+Default.args = {
+  border: 'normal',
+};
+
+export const Dark = Template.bind({});
+Dark.args = {
+  border: 'normal',
+  dark: true,
+};
```
Comment (igorizviekov) [inline src/components/InputComponent/InputComponent.stories.tsx:27]: Please add stories for error & disabled states as per the [requirement](https://github.com/igorizviekov/mt-booking_app/issues/18)

**Review by igorizviekov — CHANGES_REQUESTED** [review]
Add disabled and error states

File: src/components/InputComponent/inputComponent.types.d.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+/**
+ * Types for Input
+ * @param {string} type - Type of the input
+ */
+
+export interface IInputComponentProps extends HTMLAttributes<HTMLInputElement> {
+  border: Rounded.normal | Rounded.rounded;
+  dark: boolean;
```
Comment (igorizviekov) [inline src/components/InputComponent/inputComponent.types.d.ts:15]: add `errorMessage` and `error` props here

Comment (igorizviekov) [reply src/components/InputComponent/inputComponent.types.d.ts:15]: also, rename interface to be IInputProps

Comment (ufoBeliever) [reply src/components/InputComponent/inputComponent.types.d.ts:18]: https://github.com/igorizviekov/mt-booking_app/commit/d1ed5fdefd0af9a627399bf0ab1541da866ae781

File: src/components/InputComponent/InputComponent.stories.tsx (hunk 1 above)
Comment (ufoBeliever) [reply src/components/InputComponent/InputComponent.stories.tsx:37]: https://github.com/igorizviekov/mt-booking_app/commit/d1ed5fdefd0af9a627399bf0ab1541da866ae781

File: src/components/InputComponent/InputComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    if (inputValueReciever) {
+      inputValueReciever(value);
+    }
+  }, [value, inputValueReciever]);
+
+  return (
+    <div className='input-wrapper'>
+      <input
+        value={value}
+        onChange={inputChangeHandler}
```
Comment (igorizviekov) [inline src/components/InputComponent/InputComponent.tsx:30]: please remove `inputChangeHandler` and pass `onChange` as a prop to the component to make it reusable.
 
 Also lift state up for `value` as well, passing it as a prop

File: src/components/InputComponent/InputComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        className='input-wrapper__input'
+        disabled={disabled}
+        {...rest}
```
Comment (igorizviekov) [inline src/components/InputComponent/InputComponent.tsx:33]: always spread the props, in the end, to make sure it overrides everything else

**Review by igorizviekov — CHANGES_REQUESTED** [review]

File: src/components/InputComponent/inputComponent.types.d.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * @param {boolean} dark - Dark theme
+ * @param {boolean} disabled - Disable input
+ * @param {string} error - Set the error
+ * @param {function} inputValueReciever - Function that recieves input value
+ */
+
+export interface IInputProps extends HTMLAttributes<HTMLInputElement> {
+  border: Rounded.normal | Rounded.rounded;
+  dark: boolean;
+  disabled: boolean;
```
Comment (igorizviekov) [inline src/components/InputComponent/inputComponent.types.d.ts:20]: Isn't it included int he `HTMLInputElement` interface?

File: src/components/InputComponent/InputComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import './styles.scss';
+
+export const InputComponent: React.FC<IInputProps> = ({
+  border,
+  dark,
+  disabled,
+  error,
+  errorMessage,
+  inputValueReciever,
+  ...rest
```
Comment (igorizviekov) [inline src/components/InputComponent/InputComponent.tsx:12]: destructure `className` from it and pass it to the input className with the class name it already has like so:

const inputClassNames = ['input-wrapper__input', className].filter(Boolean).join(' ')

It is needed to add additional classes to the component if needed. please follow this approach building the components

File: src/components/InputComponent/InputComponent.tsx (hunk 3 above)
Comment (ufoBeliever) [reply src/components/InputComponent/InputComponent.tsx]: https://github.com/igorizviekov/mt-booking_app/pull/24/commits/74ff661575e673759d0bc1baf2f28de96e5ddb28

File: src/components/InputComponent/inputComponent.types.d.ts (hunk 5 above)
Comment (ufoBeliever) [reply src/components/InputComponent/inputComponent.types.d.ts]: https://github.com/igorizviekov/mt-booking_app/pull/24/commits/74ff661575e673759d0bc1baf2f28de96e5ddb28

File: src/components/InputComponent/InputComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    }
+  }, [value, inputValueReciever]);
+
+  return (
+    <div className='input-wrapper'>
+      <input
+        value={value}
+        onChange={inputChangeHandler}
+        className='input-wrapper__input'
+        disabled={disabled}
```
[the remark this answers is not in the document]
Comment (ufoBeliever) [reply src/components/InputComponent/InputComponent.tsx]: https://github.com/igorizviekov/mt-booking_app/pull/24/commits/74ff661575e673759d0bc1baf2f28de96e5ddb28

File: src/components/InputComponent/InputComponent.tsx (hunk 6 above)
Comment (ufoBeliever) [reply src/components/InputComponent/InputComponent.tsx:11]: https://github.com/igorizviekov/mt-booking_app/pull/24/commits/30f347fa0e90e34b42e8f68f6a5ffbb712b26c2d

**Review by igorizviekov — COMMENTED** [review]
Looking good

**Review by igorizviekov — APPROVED** [review]

**Pull request merged**